modules/i3s/src/types.ts

```diff
@@ -771,3 +771,141 @@ export type FieldInfo = {
   isEditable: boolean;
   label: string;
 };
+
+export type I3SWebScene = {
+  header: I3SWebSceneLayer;
+  layers: I3SWebSceneOperationalLayer[];
+};
+
+export type I3SWebSceneLayer = {
+  operationalLayers: I3SWebSceneOperationalLayer[];
+  baseMap: I3SWebSceneBaseMap;
+  ground: I3SWEbSceneGround;
+  heightModelInfo: HeightModelInfo;
+  version: string;
+  authoringApp: string;
+  authoringAppVersion: string;
+  // TODO Add presentation type. This data allows to add multiple viewes for particular web scene.
+  presentation: any;
+  initialState: I3SWebSceneInitialState;
+  spatialReference: SpatialReference;
+  viewingMode: string;
+};
+
+export type I3SWebSceneOperationalLayer = {
+  id: string;
+  opacity: number;
+  title: string;
+  url: string;
+  visibility: boolean;
+  itemId: string;
+  layerType: string;
+  layerDefinition: I3SWebSceneOperationalLayerDefinition;
+  screenSizePerspective: boolean;
+  showLabels?: boolean;
+  disablePopup?: boolean;
+  showLegend?: boolean;
+  layers?: I3SWebSceneOperationalLayer[];
+};
+
+type I3SWebSceneOperationalLayerDefinition = {
+  elevationInfo: I3SWebSceneOperationalLayerElevationInfo;
+  drawingInfo: I3SWebSceneOperationalLayerDrawingInfo;
+};
+
+type I3SWebSceneOperationalLayerElevationInfo = {
+  mode: string;
+};
+
+type I3SWebSceneOperationalLayerDrawingInfo = {
```

Comment:
Can we get a JSDoc on every type? Links to spec?

Reply:
Done